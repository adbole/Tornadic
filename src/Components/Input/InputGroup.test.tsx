import { render, screen } from "@testing-library/react";

import { Button, InputGroup } from ".";


test("Matches default snapshot", () => {
    const { container } = render(
        <InputGroup>
            <Button>Test</Button>
            <Button>Test</Button>
            <Button>Test</Button>
        </InputGroup>
    );

    expect(container).toMatchSnapshot();
});

test("Matches uniform snapshot", () => {
    const { container } = render(
        <InputGroup isUniform>
            <Button>Test</Button>
            <Button>Test</Button>
            <Button>Test</Button>
        </InputGroup>
    );

    expect(container).toMatchSnapshot();
});

test("Matches hasGap snapshot", () => {
    const { container } = render(
        <InputGroup hasGap>
            <Button>Test</Button>
            <Button>Test</Button>
            <Button>Test</Button>
        </InputGroup>
    );

    expect(container).toMatchSnapshot();
});

test("Matches hasGap and uniform snapshot", () => {
    const { container } = render(
        <InputGroup hasGap isUniform>
            <Button>Test</Button>
            <Button>Test</Button>
            <Button>Test</Button>
        </InputGroup>
    );

    expect(container).toMatchSnapshot();
});

test("Children using inputBorderRadius are given 0 radius on non-gap", () => {
    render(
        <InputGroup>
            <Button>Test</Button>
        </InputGroup>
    );

    const button = getComputedStyle(screen.getByRole("button"));

    expect(button.getPropertyValue("--input-border-radius")).toBe("0");
});
