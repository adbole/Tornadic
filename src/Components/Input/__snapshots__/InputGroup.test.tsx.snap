// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches default snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-auto-flow: column;
  margin: 5px 0px;
  border-radius: var(--input-border-radius);
  overflow: hidden;
}

.emotion-0>* {
  margin: 0;
}

.emotion-0>* {
  --input-border-radius: 0;
}

.emotion-0>*+* {
  border-left: white 1px solid;
}

.emotion-2 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  -webkit-filter: brightness(100%);
  filter: brightness(100%);
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: var(--primary);
  padding: 5px 15px;
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

.emotion-2:active,
.emotion-2:disabled {
  -webkit-filter: brightness(70%);
  filter: brightness(70%);
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
  </div>
</div>
`;

exports[`Matches hasGap and uniform snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-auto-flow: column;
  margin: 5px 0px;
  gap: 5px;
  grid-auto-columns: 1fr;
}

.emotion-0>* {
  margin: 0;
}

.emotion-2 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  -webkit-filter: brightness(100%);
  filter: brightness(100%);
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: var(--primary);
  padding: 5px 15px;
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

.emotion-2:active,
.emotion-2:disabled {
  -webkit-filter: brightness(70%);
  filter: brightness(70%);
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
  </div>
</div>
`;

exports[`Matches hasGap snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-auto-flow: column;
  margin: 5px 0px;
  gap: 5px;
}

.emotion-0>* {
  margin: 0;
}

.emotion-2 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  -webkit-filter: brightness(100%);
  filter: brightness(100%);
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: var(--primary);
  padding: 5px 15px;
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

.emotion-2:active,
.emotion-2:disabled {
  -webkit-filter: brightness(70%);
  filter: brightness(70%);
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
  </div>
</div>
`;

exports[`Matches uniform snapshot 1`] = `
.emotion-0 {
  display: grid;
  grid-auto-flow: column;
  margin: 5px 0px;
  border-radius: var(--input-border-radius);
  overflow: hidden;
  grid-auto-columns: 1fr;
}

.emotion-0>* {
  margin: 0;
}

.emotion-0>* {
  --input-border-radius: 0;
}

.emotion-0>*+* {
  border-left: white 1px solid;
}

.emotion-2 {
  -webkit-transition: filter 0.3s ease;
  transition: filter 0.3s ease;
  -webkit-filter: brightness(100%);
  filter: brightness(100%);
  border: none;
  border-radius: var(--input-border-radius);
  font-size: inherit;
  box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.25);
  margin: 5px 0px;
  color: inherit;
  background-color: var(--primary);
  padding: 5px 15px;
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

.emotion-2:active,
.emotion-2:disabled {
  -webkit-filter: brightness(70%);
  filter: brightness(70%);
}

.emotion-2:hover,
.emotion-2:focus {
  cursor: pointer;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
    <button
      class="emotion-2 emotion-3"
      type="button"
    >
      Test
    </button>
  </div>
</div>
`;
